File: go-client-sdk/pkg/models/shared/codatassessdatacontractsaccountcategoriessuggestedaccountcategorymodel.go
package shared

import (
	"time"
)

type CodatAssessDataContractsAccountCategoriesSuggestedAccountCategoryModel struct {
	DetailType   *string    `json:"detailType,omitempty"`
	ModifiedDate *time.Time `json:"modifiedDate,omitempty"`
	Subtype      *string    `json:"subtype,omitempty"`
	Type         *string    `json:"type,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatassessdatacontractsaccountcategoriessuggestedaccountcategorymodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose, Transform } from "class-transformer";


export class CodatAssessDataContractsAccountCategoriesSuggestedAccountCategoryModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "detailType" })
  detailType?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "modifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  modifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "subtype" })
  subtype?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type?: string;
}